src/metabase/api/collection.clj

```diff
@@ -82,8 +82,9 @@
 ;;; --------------------------------- Fetching a single Collection & its 'children' ----------------------------------
 
 (def ^:private valid-model-param-values
-  "Valid values for the `?model=` param accepted by endpoints in this namespace."
-  #{"card" "collection" "dashboard" "pulse" "snippet"})
+  "Valid values for the `?model=` param accepted by endpoints in this namespace.
+  `dummy` is for nilling out the set because a nil model set is actually the total model set"
+  #{"card" "collection" "dashboard" "pulse" "snippet" "dummy"})
```

Comment:
I'm reading this over and over. `#{}` means all models and we need a notion of "no models"?

Reply:
yeah, I need a non-dumb way of putting it (would highly appreciate one), but this way at least works.

so there's a tricky semantics of the collections where you need visible collections which are children of invisible collections to be visible, and look like they're at the top layer. but we want collections to never show their face in the middle collection lists, just the sidebar.

sidebar, afaict, is putting in a nil, and the middle collections list puts in a list of all models wanted except collections. there seems to be like a half-dozen other places that put in a nil and expect all models to come out. I want two things to happen:

1. no root permissions: just want the visible collections under the root collection which is invisible to show. don't want them in the middle bit.
2. root perms: show them all, but no collections in the middle bit.

previous behavior has 2 right, but not 1 because of that if-statement checking only if it's able to read, not the contents of the model.

I think the non-dummy option is definitely possible by taking away the nil set semantics and having nil set be properly nil but entails a fairly bigger change, because of those other places where the nil semantics is getting exercised.